Taric upload

xxx3456789 opened the pull request:
Full Spells, but without All of W Passives, and Champion Passive neither.
others works fine

xxx3456789 committed:
Add files via upload

xxx3456789 committed:
Add files via upload

xxx3456789 committed:
Update R.cs

xxx3456789 committed:
Update Q.cs

xxx3456789 committed:
Update Q.cs

xxx3456789 committed:
Update R.cs

danil179 commented:
Few notes before reviewing:
1. Remove commented code. 
2. Use interfaces and reduce the number of imports from GameServerLib (like AttackableUnit).
3. Remove dead code and functions that not get called. You can add them later when you will use them.

xxx3456789 committed:
Update Radiance-ally.cs

xxx3456789 committed:
Update Radiance.cs

xxx3456789 committed:
Update TaricPassive.cs

xxx3456789 committed:
Update TaricWDis.cs

xxx3456789 committed:
Update E.cs

xxx3456789 committed:
Update R.cs

xxx3456789 committed:
Update W.cs

xxx3456789 committed:
Update E.cs

xxx3456789 committed:
Update Q.cs

xxx3456789 committed:
Update R.cs

xxx3456789 committed:
Update W.cs

xxx3456789 commented:
fixed, All I have used "IMinion" "ILaneMinion" "IChampion" are changed into "IAttackableUnit"
and removed commented codes,

xxx3456789 committed:
Update E.cs

xxx3456789 committed:
Buff Stun

xxx3456789 committed:
Delete TaricPassive.cs

xxx3456789 committed:
Delete Passive.cs

xxx3456789 committed:
Update Q.cs

xxx3456789 committed:
Update W.cs

danil179 commented:
Please open an issue describing what you are adding/changing (should be something like "Adding Taric files" in this case).

danil179 commented on Champions/Taric/E.cs:
```diff
@@ -0,0 +1,72 @@
+
+            target.TakeDamage(owner, damage, DamageType.DAMAGE_TYPE_MAGICAL, DamageSource.DAMAGE_SOURCE_SPELL, false);
+
+            AddBuffHudVisual("Stun", time, 1, BuffType.COMBAT_DEHANCER, (ObjAiBase)target, time);
```
Seems that these 2 lines aren't yet ported to the API. This syntax is problematic as its using down casting and some lib functions.

xxx3456789 replied:
Well,So what can I fix it, I only know code like that, or I need to wait for API comes out (althought currently it works fine in game)

lzardy replied:
TakeDamage and AddBuffHudVisual are in the API, but AddBuffHudVisual should be moved to a buff script.

xxx3456789 replied:
Right, So only HudVisual need to moved to Buff Script, I'll fix it

xxx3456789 replied:
Changed, and I found it will be Outdated, so I just resolve it

danil179 commented on Champions/Taric/W.cs:
```diff
@@ -0,0 +1,80 @@
+using static LeagueSandbox.GameServer.API.ApiFunctionManager;
+using LeagueSandbox.GameServer.Scripting.CSharp;
+using LeagueSandbox.GameServer.GameObjects.AttackableUnits.AI;
+using LeagueSandbox.GameServer.GameObjects.Stats;
```
There should be some rework to Stats using, as in its current state there are uses of lib.

xxx3456789 replied:
about Stats, what if I use "target.Stats.~~" to code? It will be same result that use from lib, or it would be OK?

xxx3456789 replied:
Line 9 fix(with all connecting with codes) : Use "target.Stats.~~" instead of "StatModifier".

danil179 reviewed (approved):
Seems ok to me. Things will break over time (stats & visual buffs) as their rework will be done (probably in the far future).

xxx3456789 committed:
Upload Taric Stun Buff

it's using with GameBuffScript

xxx3456789 committed:
Update E.cs

change AddBuffHudVisual into AddBuffGameScript

xxx3456789 committed:
Update W (a little fix)

Line 9 fix(with all connecting with codes) : Use "target.Stats.~~" instead of "StatModifier".

xxx3456789 committed:
Update R.cs

IAttackableUnit include turrets/inhibitor/nexus, and it will cause server crash.
fixed.

xxx3456789 committed:
Update W.cs

IAttackableUnit include turrets/inhibitor/nexus, and it will crash the server.
fixed.

danil179 commented on Buffs/Taric/Radiance/Radiance.cs:
```diff
@@ -0,0 +1,35 @@
+            _statMod.AttackDamage.FlatBonus += 10f + ownerSpell.Level * 20;
+            _statMod.AbilityPower.FlatBonus += 10f + ownerSpell.Level * 20;
+            unit.AddStatModifier(_statMod);
+            var a =_healBuff = AddBuffHudVisual("Radiance", 10.0f, 1, BuffType.COMBAT_ENCHANCER, unit,10.0f);
```
These 2 lines seems to me very wrong. Why do you override _healBuff with 2 buffs to vars that never used?

xxx3456789 replied:
that's my fault, i will remove "vars" , and another thing,
if I use 2 AddBuffHudVisual(); in `_healBuff` , will it have problems?

xxx3456789 replied:
Fixed.
Removed unused vars and override problem.

danil179 commented on Buffs/Taric/TaricEhud.cs:
```diff
@@ -0,0 +1,36 @@
+            
+        }
+
+        private void OnAutoAttack(AttackableUnit target, bool isCrit)
```
Please use `IAttackableUnit` here if that's not possible inform me and I'll try to fix this.

xxx3456789 replied:
This is extra things that not needed, going to remove this part.

danil179 commented on Buffs/Taric/TaricW/TaricWDis.cs:
```diff
@@ -0,0 +1,34 @@
+            
+        }
+
+        private void OnAutoAttack(AttackableUnit target, bool isCrit)
```
Same here.

xxx3456789 replied:
going to remove this part, it was useless.

danil179 commented on Champions/Taric/R.cs:
```diff
@@ -0,0 +1,88 @@
+            foreach (var enemyTarget in GetUnitsInRange(owner, 375, true)
+                .Where(x => x.Team == CustomConvert.GetEnemyTeam(owner.Team)))
+            {
+                if (enemyTarget is IChampion || enemyTarget is IMinion && owner.Team != enemyTarget.Team)
```
Add parentheses to clarify (even if technically you don't have to).

xxx3456789 replied:
I will change it to IAttackableUnit, but it will cause to crash server, because IAttackableUnit included "Turrets/Inhibitors/Nexus".

xxx3456789 replied:
fixed to "IAttackableUnit", but still it will cause to crash server.
like i said before, it were included `Building`.

danil179 commented on Champions/Taric/R.cs:
```diff
@@ -0,0 +1,88 @@
+            foreach (var allyTarget in GetUnitsInRange(owner, 1100, true)
+                .Where(x => x.Team != CustomConvert.GetEnemyTeam(owner.Team)))
+            {
+                if (allyTarget is IChampion || allyTarget is IMinion && owner.Team == allyTarget.Team && owner != allyTarget && hasbuff == false)
```
Also here.

danil179 reviewed (changes requested):
Some comments.

xxx3456789 committed:
Update TaricEhud.cs

Remove useless part.

xxx3456789 committed:
Update TaricWDis.cs

Remove useless part.

xxx3456789 committed:
Update W.cs

`if (enemys is IChampion || enemys is IMinion && enemys != owner)` changed to `if (enemys is IAttackableUnit)`

xxx3456789 committed:
Update R.cs

1. Change "IMinion and IChampion" to "IAttackableUnit" (but still it will cause to crash the server if this part didn't changed)
2.Remove "owner.Team != enemyTarget.Team / allyTarget.Team" because it already have ".Where<>" function there.

xxx3456789 committed:
Update Radiance.cs

Fixed the problem of "unused vars" and "override using _buff" problem

danil179 commented on Buffs/Taric/Radiance/Radiance.cs:
```diff
@@ -0,0 +1,36 @@
+            _statMod.AbilityPower.FlatBonus += 10f + ownerSpell.Level * 20;
+            unit.AddStatModifier(_statMod);
+            _Buff01 = AddBuffHudVisual("Radiance", 10.0f, 1, BuffType.COMBAT_ENCHANCER, unit,10.0f);
+            _Buff02 = AddBuffHudVisual("RadianceAura", 10.0f, 1, BuffType.COMBAT_ENCHANCER, unit,10.0f);
```
I don't get why you store these in variables that never get used. Just do `AddBuffHudVisual(...);`

xxx3456789 replied:
Right, I'm dumb xD, I'll fix it.

danil179 commented on Champions/Taric/E.cs:
```diff
@@ -0,0 +1,72 @@
+            {
+                damage = 15 + spell.Level * 45 + ap * 0.3f;
+                if (owner.GetDistanceTo(target) <= 295)
+                {
```
Why is this `if` nested? should be outside for better code design (less nesting is usually better).

danil179 commented on Champions/Taric/E.cs:
```diff
@@ -0,0 +1,72 @@
+
+            ((ObjAiBase)target).AddBuffGameScript("TaricEhud", "TaricEhud", spell, time, true);
+            ((ObjAiBase)target).AddBuffGameScript("Stun", "Stun", spell, time, true);
+            var p1 = AddParticleTarget(owner, "Dazzle_tar.troy", target);
```
This `p1` never used. Remove this, or add some use for it.

danil179 commented on Champions/Taric/Q.cs:
```diff
@@ -0,0 +1,75 @@
+            if (target == owner && spell.Target == owner)
+            {
+                var Sap = owner.Stats.AbilityPower.Total * 0.42f;
+                var Sbhp = (owner.Stats.HealthPoints.Total - owner.Stats.HealthPoints.BaseValue) * 0.07f;
```
local variables shouldn't start with capital letter. Rename `Sap` , `Sbhp` to `selfAp`, `selfBaseHp`.

danil179 commented on Buffs/Taric/Radiance/Radiance-ally.cs:
```diff
@@ -0,0 +1,34 @@
+            _statMod.AttackDamage.FlatBonus += (10f + ownerSpell.Level * 20) / 2;
+            _statMod.AbilityPower.FlatBonus += (10f + ownerSpell.Level * 20) / 2;
+            unit.AddStatModifier(_statMod);
+            _healBuff = AddBuffHudVisual("Radiance", 10.0f, 1, BuffType.COMBAT_ENCHANCER, unit,10.0f);
```
Why do you need `_healBuff` variable when you never use it. Remove this var.

xxx3456789 replied:
All fixed too.

danil179 reviewed (changes requested):
All the problems are mainly code style.

I don't check the spell logic. If someone spots problems, please open an issue.

xxx3456789 committed:
Update Radiance-ally.cs

Removed `var` part.

xxx3456789 committed:
Update Radiance.cs

Removed overused `var` part.

xxx3456789 committed:
Update Q.cs

Changed the name of `Sap` & `Sbhp` to `selfAp` & `selfBaseHp`

xxx3456789 committed:
Update E.cs

Remove `nested if()` and `un-used var p1`

danil179 commented on Champions/Taric/E.cs:
```diff
@@ -0,0 +1,72 @@
+            var damage = 10 + spell.Level * 30 + ap * 0.2f;
+            if (owner.GetDistanceTo(target) <= 460)
+            {
+            damage = 15 + spell.Level * 45 + ap * 0.3f;                
```
Missing indention.

xxx3456789 replied:
All Done.

danil179 commented on Champions/Taric/E.cs:
```diff
@@ -0,0 +1,72 @@
+            }
+            if (owner.GetDistanceTo(target) <= 295)
+            {
+            damage = 20 + spell.Level * 60 + ap * 0.4f;
```
Missing indention.

danil179 commented on Champions/Taric/Q.cs:
```diff
@@ -0,0 +1,75 @@
+        private void PerformHeal(IChampion owner, ISpell spell, IAttackableUnit target)
+        {
+            var ap = owner.Stats.AbilityPower.Total * 0.3f;
+            var BaseHp = (owner.Stats.HealthPoints.Total - owner.Stats.HealthPoints.BaseValue) * 0.05f;
```
should renamed to `baseHp`.

danil179 commented on Champions/Taric/W.cs:
```diff
@@ -0,0 +1,74 @@
+            var armor = owner.Stats.Armor.Total;
+            var damage = spell.Level * 40 + armor * 0.2f;
+            var reduce = spell.Level * 5 + armor * 0.05f;
+            var p1 = AddParticleTarget(owner, "Shatter_nova.troy", owner, 1);
```
`p1` never used.

danil179 reviewed (changes requested):
Noticed new code style problems.

xxx3456789 committed:
Update E.cs

fixed indention.

xxx3456789 committed:
Update Q.cs

fixed named wrong.

xxx3456789 committed:
Update W.cs

Removed un-used var.
